### PR TITLE
chore: add independent submodules

### LINKED ISSUE #160: Introduce go submodules to use independent versioning
For projects that use multiple SDK sometimes we need to update one and stay with old version of the other. 

Imagine scenario:
v0.5.0 
- kafka-mgmt 
- registry-mgmt

Now we want received 2 changes for those apis independently but only want to use registry change. 
Users should be able to import SDK's individually with different versions

- kafka-mgmt 0.5.1
- registry-mgmt 0.5.0

### PR DESCRIPTION
**Pull request by craicoverflow**
Closes #160 

This PR introduces sub modules to give us the ability to do independent releases. This is important because the individual clients have their own release schedules and keeping them separate means that SDK consumers will only ever get a change when it is released to stage/production.

## Verification

Run the examples:

```shell
cd internal
go run ./kafkamgmt/kafka_mgmt.go
```

### CODE REVIEW

File: internal/examples/go.mod [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+replace github.com/redhat-developer/app-services-sdk-go/kafkamgmt => ../../kafkamgmt
+
+replace github.com/redhat-developer/app-services-sdk-go/kafkainstance => ../../kafkainstance
+
+replace github.com/redhat-developer/app-services-sdk-go/connectormgmt => ../../connectormgmt
+
+replace github.com/redhat-developer/app-services-sdk-go/registrymgmt => ../../registrymgmt
+
+replace github.com/redhat-developer/app-services-sdk-go/registryinstance => ../../registryinstance
```
Comment (craicoverflow) [inline internal/examples/go.mod:13]: This allows us to reference local unreleased changes in the examples.

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Releasing
```
Comment (craicoverflow) [inline RELEASING.md:1]: @secondsun @wtrocki I have added an entire section on releasing independent sub modules. Obviously this method has not been executed yet, so we may find some things that do not work when releasing the next time. Worth to have a read.

**Comment by wtrocki** [comment]
## My dream process

1. Each submodule has a changelog (as long as root). 
2. Dev does tag when wants to release SDK (releases just stub as every sdk is submodules)
3. Automation check other modules' changelogs and determines what changed. 
4. Automation will create individual tag releases for SDK
4. Changelog for each module will be published.

## Why

1. This way our release process is practically the same and there is the little cost of automating it comparing to 2-3 weeks of manual releases.

2. We can employ this strategy across fleet without problem (same bash script will work for sdk-js and sdk-java)

## Problems

Some discussion in the #166 mentioned that gitchglog might not the the best solution for changelog. I think we need to nail this one first before we invest into changelog based releases/automation.

**Comment by craicoverflow** [comment]
> Each submodule has a changelog (as long as root) that gets created and updated during PR review.

Agreed, this is described in the new release process.

> The developer writes proper change description on SDK. Notes breaking changes etc.
> This way changelog can be extended with additional info rather than relying on gitchlog that is limited

Yes this may be better to do than relying on git-chglog.

> 2. Dev does tag when wants to release SDK (releases just stub as every sdk is submodules)
> 3. Automation check other modules' changelogs and determines what changed.
> 4. Automation will create individual tag releases for SDK

Can you explain this a bit more? Where does the dev do the tag, on the root?

**Comment by wtrocki** [comment]
> Where does the dev do the tag, on the root?

Yes. 

1. Create v0.3.0 tag 
2. individual module tags are created by automation based on changelogs (changelogs will have different versions)

**Comment by craicoverflow** [comment]
> > Where does the dev do the tag, on the root?
> 
> Yes.
> 
> 1. Create v0.3.0 tag
> 2. individual module tags are created by automation based on changelogs (changelogs will have different versions)

The root module is completely independent from the submodules. Releasing the root will not include any changes from the submodules. If you create a tag for the root module then this means there are no changes in the submodules.

**Comment by wtrocki** [comment]
> The root module is completely independent from the submodules. 

Yes. It will be a nonfunctional stub used to aggregate all releases if needed. 
If someone needs to release an individual SDK that would be still possible by creating tag.

**Comment by wtrocki** [comment]
As part of the verification let's create initial releases (despite no new changes are needed)

**Pull request merged**